### PR TITLE
Reorganize CI scripts around 'GPU' and 'CPU' instead of around langua…

### PR DESCRIPTION
**Pull request by rsanthanam-amd**
…ge/features to align with upstream.

### CODE REVIEW

File: tensorflow/tools/ci_build/linux/rocm/run_cpu.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+# Force CPU
+export TF_NEED_ROCM=0
+
+yes "" | $PYTHON_BIN_PATH configure.py
+
+# Run bazel test command. Double test timeouts to avoid flakes.
+bazel test \
+      -k \
+      --test_tag_filters=-no_oss,-oss_serial,-gpu,-tpu,-benchmark-test,-v1only \
+      --test_lang_filters=py,cc \
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/linux/rocm/run_cpu.py:44]: remove this line completely?

does the upstream CI job have language filter option?

File: tensorflow/tools/ci_build/linux/rocm/run_cpu.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+      --test_sharding_strategy=disabled \
+      --test_size_filters=small,medium \
+      -- \
+      //tensorflow/... \
+      -//tensorflow/compiler/... \
+      -//tensorflow/python/integration_testing/... \
+      -//tensorflow/core/tpu/... \
+      -//tensorflow/lite/... \
+&& bazel test \
+       --config=xla \
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/linux/rocm/run_cpu.py]: Is this how they handle it in the upstream CPU job?  think `--config=xla` is true by default. unless upstream CPU job has this compound statement, collapse ours into one too

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+yes "" | $PYTHON_BIN_PATH configure.py
+
+# Run bazel test command. Double test timeouts to avoid flakes.
+bazel test \
+      --config=rocm \
+      -k \
+      --test_tag_filters=gpu \
+      --jobs=${N_BUILD_JOBS} \
+      --test_timeout 600,900,2400,7200 \
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py:47]: explicitly set number of test jobs to 1

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+export TF_NEED_ROCM=1
+export ROCM_PATH=$ROCM_INSTALL_DIR
+
+yes "" | $PYTHON_BIN_PATH configure.py
+
+# Run bazel test command. Double test timeouts to avoid flakes.
+bazel test \
+      --config=rocm \
+      -k \
+      --test_tag_filters=gpu \
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py]: --test_tag_filters=multi_gpu ?

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+export TF_NEED_ROCM=1
+export ROCM_PATH=$ROCM_INSTALL_DIR
+
+yes "" | $PYTHON_BIN_PATH configure.py
+
+# Run bazel test command. Double test timeouts to avoid flakes.
+bazel test \
+      --config=rocm \
+      -k \
+      --test_tag_filters=-no_oss,-oss_serial,-no_gpu,-no_rocm,-benchmark-test,-rocm_multi_gpu,-v1only \
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py]: add `gpu` tag?

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+      --test_lang_filters=py,cc \
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py:48]: same comment as before, w.r.t language filters...if possible, we want to do the same thing as upstream CI

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+      --run_under=//tensorflow/tools/ci_build/gpu_build:parallel_gpu_execute \
+      -- \
+      //tensorflow/... \
+      -//tensorflow/compiler/... \
+      -//tensorflow/python/integration_testing/... \
+      -//tensorflow/core/tpu/... \
+      -//tensorflow/lite/... \
+&& bazel test \
+      --config=rocm \
+      --config=xla \
```
Comment (deven-amd) [inline tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py]: same comment as before again, w.r.t having compound statements for XLA tests

File: tensorflow/tools/ci_build/linux/rocm/run_cpu.py (hunk 1 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_cpu.py:44]: actually the upstream CI job does have that exact language filter option.

File: tensorflow/tools/ci_build/linux/rocm/run_cpu.py (hunk 2 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_cpu.py]: you are correct - i will collapse in the next PR.  I verified this by running the collapsed test suite.

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py (hunk 3 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py:47]: will do.

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py (hunk 4 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py]: i don't follow.  my understanding is that the test tag filters exclude tests with the specified tag(s).  so why would we exclude multi_gpu tests in run_gpu_multi.py?

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py (hunk 5 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py]: same as previous question.

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py (hunk 7 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py]: will collapse.

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py (hunk 6 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py:48]: the upstream CI does include the language filters.

File: tensorflow/tools/ci_build/linux/rocm/run_cpu.py (hunk 1 above)
Comment (deven-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_cpu.py:44]: perfect...then we should have it too :)

File: tensorflow/tools/ci_build/linux/rocm/run_cpu.py (hunk 2 above)
Comment (deven-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_cpu.py]: > i will collapse in the next PR

do it in this PR itself..add a new commit, and squash it with the existing commit

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py (hunk 4 above)
Comment (deven-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py]: > test tag filters exclude tests with the specified tag(s)

incorrect.  they do both include and exclude

`--test_tag_filters=included_tag,-excluded_tag`

notice the `-` prefix for the `-excluded_tag`, and the lack of it in the `included_tag`

Comment (deven-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_multi.py]: https://docs.bazel.build/versions/master/command-line-reference.html#flag--test_tag_filters

File: tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py (hunk 5 above)
Comment (rsanthanam-amd) [reply tensorflow/tools/ci_build/linux/rocm/run_gpu_single.py:47]: When i tried adding the [+]gpu tag, the number of test targets decreased from 3405 to 813.  That doesn't seem right for multiple reasons.  Also, the current python, C++ scripts etc. do not have this tag.

**Review by deven-amd — APPROVED** [review]

**Pull request merged**